File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if(nums.size() == 0) return 0;
        
        priority_queue<int, vector<int>, greater<int>> pq;

        for(int i = 0; i < nums.size(); i++) {
            pq.push(nums[i]);
        }

        int curr = 1, maxCount = 1, prev = pq.top();
        pq.pop();
        while(!pq.empty()) {
            int temp = pq.top();
            pq.pop();
            if(temp == prev + 1) {
                curr++;
            } else if(temp == prev) {
                continue;
            } else {
                curr = 1;
            }
            prev = temp;
            maxCount = max(curr, maxCount);
        }

        return maxCount;
    }
};

File: 0128-longest-consecutive-sequence/0128-longest-consecutive-sequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  const numSet = new Set();
  let maxCount = 0;

  for (let i = 0; i < nums.length; i++) {
    numSet.add(nums[i]);
  }

  for (let num of numSet) {
    let count = 1;
    if (!numSet.has(num - 1)) {
      while (numSet.has(num + 1)) {
        count++;
        num++;
      }

    }

    maxCount = Math.max(maxCount, count);
  }

  return maxCount;
};